library/std/src/thread/local.rs

```diff
@@ -1137,12 +1137,20 @@ pub mod os {
         //
         // Note that to prevent an infinite loop we reset it back to null right
         // before we return from the destructor ourselves.
-        unsafe {
+        //
+        // The return value of `panic::catch_unwind` must not be dropped before
+        // `always_abort` is called, as the drop function of the value can panic
+        // itself, causing an unwinding to happen outside the `catch_unwind`
+        // call.
+        if let Err(_err) = crate::panic::catch_unwind(|| unsafe {
```

Comment:
This doesn't guard against `panic_any(DropBomb)`, which we may or may not care about here.

Reply:
The DropBomb did get dropped after the aborting panic below, or in other words it will never be dropped at all.

Reply:
Resolving conversation as I think we have an agreement here that this is not an issue.